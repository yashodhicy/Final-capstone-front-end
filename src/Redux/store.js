import { configureStore } from '@reduxjs/toolkit';
import HouseSlice from './HouseSlice';
import reducer from './reservation/reducer';


const store = configureStore({
  reducer: {
    Houses: HouseSlice,
    reservations: reducer
  },
});

export default store;
