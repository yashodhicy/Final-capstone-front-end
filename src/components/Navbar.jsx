import { useState } from "react";
//import { useDispatch } from 'react-redux';
import { Link, useLocation, useNavigate } from "react-router-dom";
import "./componentsCss/navbar.css";
import { Button } from "react-bootstrap";
import { clearSession, retrive } from "./utils/localstorage";
import axios from "axios";
import { useDispatch } from "react-redux";
import { deleteUserData } from "../Redux/user/userReducer";
import { toast } from "react-toastify";

const Navbar = () => {
  // hooks
  const path = useLocation();
  const [tray, settray] = useState(true);
  const navigate = useNavigate();
  const dispatch = useDispatch();

  const setNavigationBackground = (matchPath) => {
    if (path.pathname === matchPath)
      return { backgroundColor: "var(--green)", color: "white" };
    return {};
  };
  const navigationClass = (matchPath) => {
    return path.pathname === matchPath ? "text-white" : "";
  };

  const signout = () => {
    const url = "https://house-rental-8mh7.onrender.com/auth/sign_out";
    const userSession = retrive("session");
    axios.delete(url, { params: userSession }).then(() => {
      dispatch(deleteUserData());
      clearSession("session");
      navigate("/login");
      toast.success("You successfully signed out, see you again.");
    });
  };

  const onSignout = () => {
    const confirm = window.confirm("Do you really want to leave?");
    if (confirm) signout();
  };
  return (
    <>
      <div className="mobile-nav">
        <span
          role="button"
          onClick={() => settray(!tray)}
          onKeyDown={() => settray(!tray)}
          tabIndex={0}
        >
          <i className="fa fa-solid fa-bars fa-lg p-2" />
        </span>
      </div>
      <div
        id="navbar"
        className={tray ? "block" : "show"}
        role="button"
        onClick={() => settray(!tray)}
        onKeyDown={() => settray(!tray)}
        tabIndex={0}
      >
        <div className="top">
          <h2>HOUSE RENT</h2>
          <ul>
            <li style={setNavigationBackground("/")}>
              <Link className={navigationClass("/")} to="/">
                House
              </Link>
            </li>
            <li style={setNavigationBackground("/reserve")}>
              <Link className={navigationClass("/reserve")} to="reserve">
                Reserve
              </Link>
            </li>
            <li style={setNavigationBackground("/myreserve")}>
              <Link className={navigationClass("/myreserve")} to="myreserve">
                My Reservations
              </Link>
            </li>
            <li style={setNavigationBackground("/add")}>
              <Link className={navigationClass("/add")} to="add">
                Add House
              </Link>
            </li>
            <li>
              <Link to="delete">Delete House</Link>
            </li>
            <div>
              <Button onClick={onSignout} variant="danger">
                sign out
              </Button>
            </div>
          </ul>
        </div>
        <div className="socials">
          <ul>
            <li>
              <i className="fa fa-brands fa-twitter" />
            </li>
            <li>
              <i className="fa fa-brands fa-facebook-f" />
            </li>
            <li>
              <i className="fa fa-brands fa-pinterest-p" />
            </li>
            <li>
              <i className="fa fa-brands fa-google-plus-g" />
            </li>
          </ul>
        </div>
        <div
          className="mobile cancel"
          role="button"
          onClick={() => settray(!tray)}
          onKeyDown={() => settray(!tray)}
          tabIndex={0}
        >
          <i className="fa fa-solid fa-xmark" />
        </div>
      </div>
    </>
  );
};

export default Navbar;
