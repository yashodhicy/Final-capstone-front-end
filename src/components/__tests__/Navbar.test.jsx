import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../../Redux/store';
import Navbar from '../Navbar'

describe('Test Navbar Component', () => {
    test ('it renders correctly', () => {
        const tree = renderer
        .create(
            <Provider store={store}>
                <Router>
                    <Navbar />
                </Router>
            </Provider>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    })
});
