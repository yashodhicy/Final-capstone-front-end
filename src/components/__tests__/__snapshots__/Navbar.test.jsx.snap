// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Navbar Component it renders correctly 1`] = `
[
  <div
    className="mobile bars"
  >
    <span
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <i
        className="fa-solid fa-bars"
      />
    </span>
  </div>,
  <div
    className="block"
    id="navbar"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="top"
    >
      <h2>
        HOUSE RENT
      </h2>
      <ul>
        <li
          style={
            {
              "backgroundColor": "yellowgreen",
              "color": "white",
            }
          }
        >
          <a
            className="text-white"
            href="/"
            onClick={[Function]}
          >
            House
          </a>
        </li>
        <li
          style={{}}
        >
          <a
            className=""
            href="/reserve"
            onClick={[Function]}
          >
            Reserve
          </a>
        </li>
        <li
          style={{}}
        >
          <a
            className=""
            href="/myreserve"
            onClick={[Function]}
          >
            My Reservations
          </a>
        </li>
        <li
          style={{}}
        >
          <a
            className=""
            href="/add"
            onClick={[Function]}
          >
            Add House
          </a>
        </li>
        <li>
          <a
            href="/delete"
            onClick={[Function]}
          >
            Delete House
          </a>
        </li>
        <div>
          <button
            className="btn btn-danger"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            sign out
          </button>
        </div>
      </ul>
    </div>
    <div
      className="socials"
    >
      <ul>
        <li>
          <i
            className="fa-brands fa-twitter"
          />
        </li>
        <li>
          <i
            className="fa-brands fa-facebook-f"
          />
        </li>
        <li>
          <i
            className="fa-brands fa-pinterest-p"
          />
        </li>
        <li>
          <i
            className="fa-brands fa-google-plus-g"
          />
        </li>
      </ul>
    </div>
    <div
      className="mobile cancel"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <i
        className="fa-solid fa-xmark"
      />
    </div>
  </div>,
]
`;
